### PR TITLE
#822 gRPC layer removal: adaptations of the prober tests

### PR DESCRIPTION
**Pull request by mickmis**
Seventh batch of modifications, those changes enable all tests to be validated with the new implementation.

- change from POST to GET the method to retrieve the token: because the generated code used to not check for the HTTP method, the POST call used to work, while this was actually not according to the OpenAPI definition of the dummy oauth provider
- aux API: do not retrieve anymore the version through the `as_string` subfield, which is a construct inherited from gRPC
- in all APIs: remove the error message exact comparison: now we have an error ID embedded, so we only do a substring check
- in all APIs: some validation messages changed due to the fact that it is not possible anymore for some fields to be missing (when they are specified as required they will be present but empty in the go models)
- in SCD API: some calls for creating resources used to return 200 when success, while the specs specify 201
- in RID v2 API: adapt scopes used in some endpoints, which were not implemented as per specs in the previous implementation (please check that this is indeed correct and not something that was intended as such but not specified properly like the scope issue for RID v1 API we has discussed)

After this, 2 more upcoming PRs before everything is done (modulo if I forget something): 1. fix for the path prefix; 2. cleanup and documentation

### CODE REVIEW

File: monitoring/prober/rid/v2/test_isa_expiry.py [hunk 1]
```diff
@@ -14,7 +14,7 @@
 
 
 def test_ensure_clean_workspace_v2(ids, session_ridv2):
-  resp = session_ridv2.get('{}/{}'.format(ISA_PATH, ids(ISA_TYPE)), scope=SCOPE_SP)
+  resp = session_ridv2.get('{}/{}'.format(ISA_PATH, ids(ISA_TYPE)), scope=SCOPE_DP)
```
Comment (BenjaminPelletier) [inline monitoring/prober/rid/v2/test_isa_expiry.py:17]: This is a bug in the OpenAPI -- Service Providers should be able to retrieve ISA information.  That change is now in `updated.yaml`, so the uastech/standards subrepo can be updated to [here](https://github.com/uastech/standards/commit/dd4016b09fc8cb98f30c2a17b5a088fb2995ab54) (and this should stay as `SCOPE_SP`)

File: monitoring/prober/rid/v2/test_isa_simple.py [hunk 2]
```diff
@@ -80,7 +80,8 @@ def test_get_isa_by_id(ids, session_ridv2):
 
 @default_scope(SCOPE_SP)
 def test_update_isa(ids, session_ridv2):
-  resp = session_ridv2.get('{}/{}'.format(ISA_PATH, ids(ISA_TYPE)))
+  resp = session_ridv2.get('{}/{}'.format(ISA_PATH, ids(ISA_TYPE)), scope=SCOPE_DP)
```
Comment (BenjaminPelletier) [inline monitoring/prober/rid/v2/test_isa_simple.py:83]: Same as test_is_expiry.py -- this should stay as SCOPE_SP after `updated.yaml` is updated

File: monitoring/prober/scd/test_constraint_simple.py [hunk 3]
```diff
@@ -108,7 +108,7 @@ def test_create_constraint(ids, scd_api, scd_session):
   assert resp.status_code == 403, resp.content
 
   resp = scd_session.put('/constraint_references/{}'.format(id), json=req, scope=SCOPE_CM)
-  assert resp.status_code == 200, resp.content
+  assert resp.status_code == 201, resp.content
```
Comment (BenjaminPelletier) [inline monitoring/prober/scd/test_constraint_simple.py:111]: I'm looking forward to finally fixing the category of problem from #462 :)  (the 201 is a solid improvement)

File: monitoring/prober/rid/v2/test_isa_simple.py (hunk 2 above)
Comment (mickmis) [reply monitoring/prober/rid/v2/test_isa_simple.py]: Now fixed

File: monitoring/prober/rid/v2/test_isa_expiry.py (hunk 1 above)
Comment (mickmis) [reply monitoring/prober/rid/v2/test_isa_expiry.py]: Now fixed

**Pull request merged**